[Yiwen Lai] iP

SeaHogs opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

SeaHogs committed:
Level-0

SeaHogs committed:
Level-1

SeaHogs committed:
Level-1

SeaHogs committed:
Level-2

SeaHogs committed:
Merge remote-tracking branch 'origin/master'

SeaHogs committed:
Level-2

SeaHogs committed:
Level-2

SeaHogs committed:
Level-2

SeaHogs committed:
Level-3

SeaHogs committed:
Level-3

SeaHogs committed:
Level-4

SeaHogs committed:
A-TextUiTesting

SeaHogs committed:
A-TextUiTesting

SeaHogs committed:
Level-5

SeaHogs committed:
Level-6

SeaHogs committed:
Level-6

SeaHogs committed:
Level-6

SeaHogs committed:
Level-7

SeaHogs committed:
Level-7

SeaHogs committed:
Fix a bug with Deadline string manipulation.

SeaHogs committed:
Fixed a bug where delete and mark is not saved to the txt file.
Fixed a bug where save in LocalList is not changing what is in the txt file

SeaHogs committed:
Fixed a bug where delete and mark is not saved to the txt file.
Fixed a bug where save in LocalList is not changing what is in the txt file

SeaHogs committed:
Different duke.txt

SeaHogs committed:
Added LocalDate and LocalTime

String value date stored in Activity was swapped for LocalDate and LocalTime. This allows the date and time data to be compared and formatted through different way.

SeaHogs committed:
New commit message

SeaHogs committed:
Added LocalDate and LocalTime, changed structure of constructor, and fixed bug with save and load utility.

String value date stored in Activity was swapped for LocalDate and LocalTime. This allows the date and time data to be compared and formatted through different way. The constructor of Activity subclass have been changed from single string input to serial string each presenting from name to finish time for easier creation when loading from file. Fixed bug with save and load reading wrong number of data causing index out of bond exception.

SeaHogs committed:
Merge remote-tracking branch 'origin/master'

SeaHogs committed:
Change in file content
File content changed as the program runs due to save and load function

SeaHogs committed:
No changes
Intellij somewhat identifies there a change and will not allow me to fork add-gradle-support until I commit this again.

SeaHogs committed:
Minor coding standard fixes
Fixed a few indentation error within the Duke.java file

SeaHogs committed:
Set up gradle
Copied packages from master branch and modified mainClass.set() in build.gradle so my version of Duke.java can run on gradle.

SeaHogs committed:
Merge branch 'add-gradle-support'

SeaHogs committed:
Added invalid time handling
Added tests for time when the date is either null or unreasonable. When such condition happens, the constructor will throw Runtime exception. The input is regarded as true only when all date and time are not null and start date and time is before end date and time.

SeaHogs committed:
Changed error message
Edited the error message from saying an error occurred to more specific information about file missing, making debugging later easier.

SeaHogs committed:
Created duke.txt
This txt file is used for temporary storage of the data produced by EventTest.java. The file will be emptied each time the test finish running.

SeaHogs committed:
Add unit tests for Event class methods in Duke project

- Implement `setUp` method with `@BeforeEach` annotation to initialize an Event instance before each test.
- Add `testMarkAsDone` to verify marking an event as done updates its status correctly.
- Add `testMarkAsUndone` to ensure marking an event as undone reverts its status appropriately.
- Introduce `testExceptionMessage` to confirm throwing of RuntimeException with a specific message when an event's end date is set before its start date. This test ensures the Event constructor validates date inputs correctly.

These tests aim to increase the Event class's reliability and maintainability by validating its core functionalities and error handling mechanisms.

SeaHogs committed:
Add unit tests for LocaList class methods in Duke project

- Implement `setUp` method with `@BeforeEach` annotation to initialize an LocalList instance before each test.
- Add `testSaveActivities` to verify if saving and loading data into the text file duke.txt is done correctly.
- Add `testInvalidFile` to ensure LocalList is handling instance where file path is invalid appropriately.
- Implemented 'tearDown' method with 'AfterEach' annotation to reset the content of duke.txt file to empty.

These tests aim to increase the LocalList class's reliability and maintainability by validating its core functionalities and error handling mechanisms.

SeaHogs committed:
Created directory java under main/test directory
 -created to store JUnit test file
 -currently stores EventTest.java and LocalListTest.java under Duke.activityAndUtility

SeaHogs committed:
Add comprehensive Javadoc documentation across classes

This commit introduces detailed Javadoc documentation for the Activity, Deadline, Event, Todo, ActivityList, Duke, LocalList, EventTest, and LocalListTest classes within the Duke application. Each class and its methods now have clear descriptions, including the purpose, parameters, and expected outcomes. This documentation enhances code readability, facilitates easier maintenance, and improves the onboarding process for new developers working on the project.

SeaHogs committed:
Modified indentation and comments

Modified indentation in DateTimeFormat, Deadline, Event and Duke java file to better fit coding standard. Removed some unused comment and added some comment for future comprehension.

SeaHogs committed:
Add searchList class for substring search and activity printing

 -Introduce the searchList class to the Duke.activityAndUtility package, providing utilities for searching through a list of strings for substrings and printing activities from a corresponding list based on the search results.
 -The class includes the findIndicesOfSubstring method for locating all indices of strings containing a specified substring and the printList method for displaying activities related to these search terms. -
 -This addition aims to enhance the application's ability to filter and display data based on user-defined criteria.
 -javaDoc included in searchList

SeaHogs committed:
Implement findActivity method in activity management

 -Add a new method, findActivity, to enhance search functionality within the application.
 -This method leverages the searchList class to search through a predefined list of activity names or descriptions for a given substring and print the matching activities. It simplifies the process of finding specific activities for users by abstracting the search and display logic into a single, accessible method.
 -condition in main also been modified to check for first word "find" as trigger for findActivity.

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/Activity.java:
```diff
@@ -0,0 +1,29 @@
+ * It provides methods for printing details about an activity, retrieving the name of the activity, and marking an activity
+ * with a specific input, potentially to track completion or progress.
+ */
+public interface Activity {
```
Good use of interface here

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/DateTimeFormat.java:
```diff
@@ -0,0 +1,55 @@
+ */
+class DateTimeFormat {
+    /**
+     * Parses a {@link LocalDate} from the provided input string. The method supports two date formats: "yyyy-MM-dd" and "dd-MM-yyyy".
+     * It attempts to parse the date using the first format and falls back to the second format if the first attempt fails.
+     *
+     * @param input The input string containing the date to be parsed. The date is expected to be the first part of the input,
+     *              separated by a space from any other information.
+     * @return The parsed {@link LocalDate}, or {@code null} if neither format can parse the input successfully.
```
Not sure if its on my end, but the spacing seems inconsistent as seen in other comments as well.

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/DateTimeFormat.java:
```diff
@@ -0,0 +1,55 @@
+        //checking if it fit either format
+        try {
+            return LocalDate.parse(dateString[0], formatter1);
+        } catch (DateTimeParseException ignored) {
```
Works but perhaps try not to use try-catch for control flow.

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/DateTimeFormat.java:
```diff
@@ -0,0 +1,55 @@
+        try {
+            return LocalDate.parse(dateString[0], formatter2);
+        } catch (DateTimeParseException format) {
+            return null;
```
Returning null is not recommended in my opinion, perhaps consider throwing a exception from the function instead and catching it.

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/Deadline.java:
```diff
@@ -0,0 +1,82 @@
+ * printed, named, and marked as complete or incomplete.
+ */
+public class Deadline implements Activity {
+    List<String> act;
```
I think a better use of OOP is to abstract out the List management to another TaskManager class.

Deadline, event and todo should be just classes to represent a deadline, .. object.

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/Event.java:
```diff
@@ -0,0 +1,109 @@
+     */
+    @Override
+    public void printActivity() {
+        if (startDate != null && startTime != null && endDate != null && endTime != null) {
```
null check is not recommended could cause problems in swe
https://stackoverflow.com/questions/1274792/is-returning-null-bad-design

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/LocalList.java:
```diff
@@ -0,0 +1,89 @@
+                        break;
+                    case "D":
+                        activity = new Deadline(parts[1], parts[2], parts[3]);
+                        break;
```
Good use of Loader class

UdhayaShan1 commented on src/main/java/Duke/activityAndUtility/searchList.java:
```diff
@@ -0,0 +1,54 @@
+        List<Integer> indices = findIndicesOfSubstring(searchTable, substr); //get the index of required string
+        if (!indices.isEmpty()) {
+            System.out.format("\tHere's where I have: \n");
+            System.out.println("\t____________________________________________________________");
```
Perhaps can abstract the line to a static field.

UdhayaShan1 commented on src/test/java/Duke/activityAndUtility/EventTest.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    /**
+     * Tests marking an event as not done. It verifies that the status of the event is correctly updated to "X".
```
Good comprehensive tests.

UdhayaShan1 reviewed (commented):
Looks good.
Recommend some changes to OOP design.

SeaHogs committed:
Rearranged structure for OOP

 -created package command replacing functions in ActivityList
 -created AddActivity, AddEvent, AddTodo, DeleteActivity, FindActivity, MarkActivity
 -moved input handling from Duke to phrase

SeaHogs committed:
Add GUI to Duke project

The Duke project lacks a graphical user interface (GUI), making it less user-friendly.
To improve the overall user experience and make Duke more accessible, it is essential to implement a GUI.
Implement the GUI by adding a DialogBox for user interactions, a Launcher for application startup, and a MainWindow to host the GUI components.

SeaHogs committed:
Add test cases

The test cases temper during A-MoreOOP commit.
Recreate test cases LocalListTest and DateFormatTest.

SeaHogs committed:
Add assertion

Add assert in LocalList
ensure that the file is created and read successfully.
If file creation fails, an assertion failure will be triggered.

Add assert in phrase
ensure that string input is not null
encase this is not captured by Unknow.java

SeaHogs committed:
Merge pull request #2 from SeaHogs/branch-A-Assertions

Add assertion

SeaHogs committed:
Add javadoc

Add javadoc for all class in Activities package

SeaHogs committed:
Add javadoc
Remove unused import

Add javadoc for all class in Commands package.
Remove unused import in both Commands package and Activities package.

SeaHogs committed:
Add javadoc
Remove unused import

Add javadoc for all class in Conversation, duke, Exception, LocalStorage
phrase, UI package.
Remove unused import.

SeaHogs committed:
Merge pull request #3 from SeaHogs/branch-A-CodeQuality

Branch-A-Code-Quality

SeaHogs committed:
Add detect duplicate

Can detect duplicate when adding activity.
Throws CommandException with error message when detected.

SeaHogs committed:
Edit duke.txt

SeaHogs committed:
Bug fix: UI

During string overrun in DialogBox,
string will be cropped.
Updated DialogBox to have dynamic size.

SeaHogs committed:
Change Ui colour

Change the DialogBox colour for user to white,
change the DialogBox colour for client to yellow.

SeaHogs committed:
fix: index out of bonds for all space input

If input containing only space,
cause index of bonds for String firstword in phrase.
Add condition to check for all space input.

SeaHogs committed:
Add userguide

Modify README.md for userguide

SeaHogs committed:
Add LocalStorage to README.md

Add description of LocalStorage to userguide.

SeaHogs committed:
Modify testCases

Fixed LocalListTest's condition to sync with new output prompt.

SeaHogs committed:
fix: build.gradle

Fix error in mainClass name and jar name in build.gradle.

SeaHogs committed:
fix: build.gradle

Fix error in duplicate setMainClass in build.gradle.

SeaHogs committed:
Update download link

Change the download link to the lastest release link.

SeaHogs committed:
Add Ui image

SeaHogs committed:
Add title name

Add title name to make product name visible in the title bar of the Window.

SeaHogs committed:
Add Ui.png

SeaHogs committed:
fix: Java version

Fix gradle causing jar to have java.lang.UnsupportedClassVersionError.

SeaHogs committed:
fix: JavaFx version

Fix gradle causing jar to have java.lang.UnsupportedClassVersionError.

SeaHogs committed:
fix: README.md
Fix format in readme where it only renders correctly in intellij and github pages but not deployed github pages.

SeaHogs committed:
fix: README.md
Fix format in readme where it only renders correctly in intellij and github pages but not deployed github pages.

SeaHogs committed:
fix: README.md
Fix format in readme where it only renders correctly in intellij and github pages but not deployed github pages.

SeaHogs committed:
Update README.md
